packages/cardano-services/test/Asset/TypeOrmNftMetadataService.test.ts

```diff
@@ -0,0 +1,52 @@
+import { Cardano } from '@cardano-sdk/core';
+import { TypeOrmNftMetadataService, createDnsResolver, getConnectionConfig, getEntities } from '../../src';
+import { logger, mockProviders } from '@cardano-sdk/util-dev';
+
+describe('TypeOrmNftMetadataService', () => {
+  let service: TypeOrmNftMetadataService;
+
+  beforeAll(async () => {
+    const dnsResolver = createDnsResolver({ factor: 1.1, maxRetryTime: 1000 }, logger);
+    const entities = getEntities(['asset']);
+    const connectionConfig$ = getConnectionConfig(dnsResolver, 'test', 'Handle', {
+      postgresConnectionStringHandle: process.env.POSTGRES_CONNECTION_STRING_HANDLE!
+    });
+    service = new TypeOrmNftMetadataService({ connectionConfig$, entities, logger });
+    await service.initialize();
+    await service.start();
+  });
+
+  afterAll(async () => {
+    jest.restoreAllMocks();
+    await service.shutdown();
+  });
+
+  it('should return null for a non-existing asset', async () => {
+    const nfMetadata = await service.getNftMetadata({
+      name: mockProviders.handleAssetInfo.name,
+      policyId: mockProviders.handleAssetInfo.policyId
+    });
+
+    expect(nfMetadata).toBe(null);
+  });
+
+  it('should retrieve metadata for an existing nft', async () => {
```

Comment:
We're missing some test coverage. I suggest to update the existing test to just required properties, then add add additional test(s) for optional ones. If you can't find NFTs with those optional properties in the database then you can insert them in test setup.
```suggestion
  describe('existing nft', () => {
    it('should retrieve files when they are present');
    it('should retrieve mediaType when it's present');
    it('should retrieve otherProperties when they're present');
    it('should retrieve description when it's present');
    it('should retrieve name, image and version (all required properties)', async () => {
```

Reply:
Fixed on 740c602 and 8d7d0ef

Reply:
Need to add update e2e test to have a reproducible source of 8d7d0efa2a965854c5192609f15206f881cdc65b

Reply:
Fixed on be37992